### PR TITLE
Codecov & AppVoyer improvements

### PR DESCRIPTION
**Pull request by dimoschi**
An attempt to enable codecov

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -6,7 +6,10 @@ whitelist_externals = source
...
 setenv = TOX_ENV_NAME={envname}
-passenv = TOX_* TRANSIFEX_USER TRANSIFEX_PASSWORD CI_* APPVEYOR_*
+passenv = TOX_* TRANSIFEX_USER TRANSIFEX_PASSWORD CI CI_* CIRCLECI CIRCLE* APPVEYOR*
+deps = codecov>=1.4.0
+       coverage
 commands = python -V
-           python setup.py test
+           coverage run --append setup.py test
            bash ./contrib/test_build.sh
+           codecov -e TOX_ENV_NAME --token=5ad93ecd-ffdc-4c92-b2db-a7c0ff1c1571
```
Comment (kouk) [inline tox.ini]: @dimoschi shouldn't we put the codecov token in the CircleCI environment? And maybe disable codecov when running on AppVeyor?

**Comment by kouk** [comment]
Thanks @stevepeak for your help. Codecov is great btw.

**Comment by kouk** [comment]
@dimoschi could you fix the issue with the token I mentioned above?

**Comment by dimoschi** [comment]
@kouk tests fail because of the PO files. Transifex doesn't accept anymore PO files without a msgstr. However I looked old appveyor tests and I haven't found codecov being run inside those tests (maybe I've missed something).

**Comment by dimoschi** [comment]
@kouk I don't see codecov running in appveyor, so I am not sure what do you mean with your comment. Also we only need to run appveyor only in master branch. That's because appveyor merges the PR and uses `FERCHED_HEAD` to run the build. So it builds master + the PR, which I think is the desired outcome (I think)

**Comment by dimoschi** [comment]
@kouk Another point is I am trying to export an xml report, so CircleCI can gather data but I get that error: `ERROR: InvocationError: '/home/ubuntu/transifex-client/.tox/py35/bin/coverage xml -o /tmp/circle-junit.2pBrDmk/django/output.xml'`

**Comment by dimoschi** [comment]
@kouk This could be merged. It enables codecov for tx-client. Also, it runs Appveyor only for master branch. I'm quoting from my previous comment:

> Also we only need to run appveyor only in master branch. That's because appveyor merges the PR and uses FERCHED_HEAD to run the build. So it builds master + the PR, which I think is the desired outcome (I think).

**Pull request merged**